replica/database.cc

```diff
@@ -1422,6 +1423,15 @@ void database::validate_new_keyspace(keyspace_metadata& ksm) {
     if (has_keyspace(ksm.name())) {
         throw exceptions::already_exists_exception{ksm.name()};
     }
+    std::visit(overloaded_functor {
+        [] (const storage_options::local&) {
+        },
+        [this] (const storage_options::s3& so) {
+            if (!_feat.keyspace_storage_options) {
+                throw exceptions::invalid_request_exception("Keyspace storage options not supported in the cluster");
+            }
+        }
+    }, ksm.get_storage_options().value);
```

Comment:
The function name validate_new_keyspace() suggests (?) it only applies to CREATE KEYSPACE, not to ALTER KEYSPACE. Can you please add to your test also a check what happens if you try to ALTER a keyspace to use S3? Is this even possible - even if S3 *is* configured?

Reply:
> Can you please add to your test also a check what happens if you try to ALTER a keyspace to use S3? 

Changing keyspace storage type is not allowed:

```
# Test that storage options cannot be altered (at least until it's well defined
# what it means to e.g. switch from S3 to another format and back).
def test_storage_options_alter_type(cql, scylla_only):
    ksdef = "WITH REPLICATION = { 'class' : 'NetworkTopologyStrategy', 'replication_factor' : '1' } " \
            "AND STORAGE = { 'type' : 'LOCAL' }"
    with new_test_keyspace(cql, ksdef) as keyspace:
         # It's not fine to change the storage type
        ksdef_local = "WITH REPLICATION = { 'class' : 'NetworkTopologyStrategy', 'replication_factor' : '1' } " \
            "AND STORAGE = { 'type' : 'S3', 'bucket' : '/b1', 'endpoint': 'localhost'}"
        with pytest.raises(InvalidRequest):
            res = cql.execute(f"ALTER KEYSPACE {keyspace} {ksdef_local}")
```

Reply:
Thanks.